.gitignore

```diff
@@ -92,3 +92,44 @@ docs/html/
 
 # Python
 __pycache__/
+
+# Object Files
+*.o
+*.ko
+*.obj
+*.elf
+
+# Precompiled Headers
+*.gch
+*.pch
+
+# Libraries
+*.lib
+*.a
+*.la
+*.lo
+
+# Shared objects (inc. Windows DLLs)
+*.dll
+*.so
+*.so.*
+*.dylib
+
+# Executables
+*.exe
+*.out
+*.app
+*.i*86
+*.x86_64
+*.hex
+
+# Debug files
+*.dSYM/
+
+ta_client
+/endpoint/build/
```

Comment:
We should have consistent output directory to keep the generated endpoint executable file(s).